frontend/packages/dev-console/src/components/monitoring/MonitoringPage.tsx

```diff
@@ -11,20 +13,52 @@ import MonitoringEvents from './events/MonitoringEvents';
 
 export const MONITORING_ALL_NS_PAGE_URI = '/dev-monitoring/all-namespaces';
 
-interface MonitoringPageProps {
+type MonitoringPageProps = {
   match: RMatch<{
     ns?: string;
   }>;
-}
+};
+
+type StateProps = {
+  canAccess: boolean;
+};
+
+type Props = MonitoringPageProps & StateProps;
 
 const handleNamespaceChange = (newNamespace: string): void => {
   if (newNamespace === ALL_NAMESPACES_KEY) {
     history.push(MONITORING_ALL_NS_PAGE_URI);
   }
 };
 
-export const MonitoringPage: React.FC<MonitoringPageProps> = ({ match }) => {
+export const MonitoringPage: React.FC<Props> = ({ match, canAccess }) => {
   const activeNamespace = match.params.ns;
+  const canAccessPrometheus = canAccess && !!window.SERVER_FLAGS.prometheusBaseURL;
+
+  const pages = React.useMemo(
```

Comment:
Not sure we needed a useMemo here as nothing here is complex, but I see no harm.

Reply:
Just didn't wanted to cause any unnecessary re-renders because change in `pages`.